### PR TITLE
Workbench: Project recovery progress bar fix

### PR DESCRIPTION
**Pull request by Pasarus**
**Description of work.**
Due to git merging some code poorly, the connections between the model and view of the PythonFileInterpreter were not being made, this in turn lead to the recovery view never being called by the emit.

**To test:**
- Open Workbench
- Run this script:
```python
for ii in range(0, 1000):
    CreateSampleWorkspace(OutputWorkspace=str(ii))
```
- Wait for recovery to save (Shown in the messages tab as `Project Recovery: Saving finished`)
- Run: `Segfault(0)`
- Open Workbench
- Recovery last checkpoint
- Notice the progress bar works.

*There is no associated issue.*

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/applications/workbench/workbench/projectrecovery/recoverygui/projectrecoverymodel.py [hunk 1]
```diff
@@ -103,6 +103,7 @@ def recover_selected_checkpoint(self, selected):
             if isinstance(e, KeyboardInterrupt):
```
Comment (DanNixon) [inline qt/applications/workbench/workbench/projectrecovery/recoverygui/projectrecoverymodel.py]: I know it's not modified in this PR but this code is pointless, `KeyboardInterrupt` subclasses `BaseException` not `Exception` so `isinstance(e, KeyboardInterrupt)` is always `False`.

(see https://docs.python.org/3/library/exceptions.html)

Comment (Pasarus) [reply qt/applications/workbench/workbench/projectrecovery/recoverygui/projectrecoverymodel.py]: In Python 2.7 which we are currently running on. It is derived from StandardError which is in turn derived from Exception.

(See https://docs.python.org/2/library/exceptions.html)

Comment (Pasarus) [reply qt/applications/workbench/workbench/projectrecovery/recoverygui/projectrecoverymodel.py]: You are not wrong.

**Comment by Pasarus** [comment]
@DanNixon Would it be possible to review this or should I remove you so someone else can do it?

**Review by soininen — APPROVED** [review]
Project recovery seems to be broken in `release-next` in more ways than just non-working progress bar. This seems to fix those issues as well as the progress bar. I think this is good to go.

**Comment by Pasarus** [comment]
A recent change to some functions that recovery relied on broke, it under certain circumstances but by default these things are caught and removed from view, for the sake of the end user.

**Pull request merged**